### PR TITLE
fix(server): createInputMiddleware next call type

### PR DESCRIPTION
**Pull request by ixahmedxi**
Closes #

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/server/src/core/middleware.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ export type MiddlewareFunction<
     meta: TParams['_meta'] | undefined;
     next: {
       (): Promise<MiddlewareResult<TParams>>;
-      <$Context>(opts: { ctx: $Context }): Promise<
+      <$Context>(opts: { ctx?: $Context; input?: unknown }): Promise<
```
Comment (KATT) [inline packages/server/src/core/middleware.ts:140]: I am about 99.9% I tried this but it had some edge-case.... but I can't remember now and it evidently passes tests 🤔 

Can you make sure it works if you downgrade to TS 4.7?

Comment (ixahmedxi) [reply packages/server/src/core/middleware.ts:140]: for sure, i shall do that :)

Comment (ixahmedxi) [reply packages/server/src/core/middleware.ts:140]: @KATT so I downgraded typescript to 4.7.4 and i found the current error in the `fastifyRequestHandler.ts` file:

image

this is however not connected to the following change of this PR as that I can confirm the error shows up on the base `next` branch. I was able to fix the error by doing the following:

image

So in conclusion it doesn't seem like my change does produce any type errors on Typescript version `4.7.4` and in the process I discovered an unrelated error that's easily solveable.

File: packages/server/src/core/internals/procedureBuilder.ts [hunk 2]
```diff
@@ -330,7 +330,10 @@ function createProcedureCaller(_def: AnyProcedureBuilderDef): AnyProcedure {
           rawInput: opts.rawInput,
           meta: _def.meta,
           input: callOpts.input,
-          next: async (nextOpts?: { ctx: any; input?: any }) => {
+          next: async (nextOpts?: {
+            ctx?: { [key: string]: unknown };
```
Comment (KATT) [inline packages/server/src/core/internals/procedureBuilder.ts:334]: This could be written with a `Record<string, unknown>`, a bit neater

**Review by KATT — APPROVED** [review]
